app/uk/gov/hmrc/eusubsidycompliancefrontend/services/EmailVerificationService.scala

```diff
@@ -102,6 +102,23 @@ class EmailVerificationService @Inject() (
     }
   }
 
+  private def generateEmailServiceUrl(redirectUrl: String)(implicit req: AuthenticatedEnrolledRequest[AnyContent]): String = {
+    if(req.isLocal()) {
+      servicesConfig.baseUrl("email-verification-frontend") + redirectUrl
+    } else {
+      redirectUrl
+    }
+  }
+
+  private def generateRedirectToSelfUrl(redirectUrl: String)(implicit req: AuthenticatedEnrolledRequest[AnyContent]): String = {
+    if(req.isLocal()) {
+      "http://" + req.host + redirectUrl
```

Comment:
It might be an idea to use `Uri` to construct the URL since it will do the right thing wrt encoding etc..